masonry_core/src/app/render_root.rs

```diff
@@ -287,6 +306,80 @@ pub(crate) struct InspectorState {
 }
 
 impl RenderRoot {
+    /// Compute a focus fallback according to the provided `policy`.
+    ///
+    /// Returns `Some(id)` if a suitable widget is found, otherwise `None`.
+    pub fn compute_focus_fallback(&self, policy: FocusFallbackPolicy) -> Option<WidgetId> {
+        let base = self.layer_root_id(0);
+        match &policy {
+            FocusFallbackPolicy::AnchorSubtreeFirst(inner) => {
+                if let Some(anchor) = self.global_state.focus_anchor
+                    && self.is_still_interactive(anchor)
+                    && let Some(id) = self.match_policy_at(inner, anchor)
+                {
+                    return Some(id);
+                }
```

Comment:
I wonder if this should also search through the entire layer the anchor is part of.
There are also arguments for having it also do a "nearest parent text-input" (or focusable).

Neither of these are blocking for this PR, though.